packages/core/cache/src/Cache.js

```diff
@@ -105,4 +76,35 @@ export class Cache {
   }
 }
 
-export default new Cache();
+// Cache for whether a cache dir exists
+const existsCache: Set<FilePath> = new Set();
+const cacheByDir: DefaultMap<FilePath, Cache> = new DefaultMap(dir => {
+  return new Cache(dir);
+});
+
+export async function createCacheDir(dir: FilePath): Promise<void> {
+  if (existsCache.has(dir)) {
```

Comment:
This existed before, and seems questionable. 

The directory could easily be removed from underneath Parcel between runs. Since `mkdirp` is safe to rerun, and it is now parallelized below, I wonder what the perf cost would be to just run this every time Parcel is about to perform a build.

Reply:
It wouldn't make any significant performance impact, just wrote a bunch of test scripts that test this. (all using fs-extra as it has a bunch of other fancy ways to do this)

cached
```JS
const fs = require("fs-extra");
const path = require("path");
const { TARGET_DIR, FILE_PATH, TEST_STRING, RUNS } = require("./constants");

let existsCache = new Set();
async function createDir() {
  if (existsCache.has(TARGET_DIR)) return;
  await fs.mkdirp(TARGET_DIR);
  existsCache.add(TARGET_DIR);
}

async function run() {
  for (let i = 0; i < RUNS; i++) {
    await createDir();
    await fs.writeFile(FILE_PATH, TEST_STRING);
  }
}

run();
```

mkdirp
```JS
const fs = require("fs-extra");
const path = require("path");
const { TARGET_DIR, FILE_PATH, TEST_STRING, RUNS } = require("./constants");

async function run() {
  for (let i = 0; i < RUNS; i++) {
    await fs.mkdirp(TARGET_DIR);
    await fs.writeFile(FILE_PATH, TEST_STRING);
  }
}

run();
```

outputFile
```JS
const fs = require("fs-extra");
const path = require("path");
const { TARGET_DIR, FILE_PATH, TEST_STRING, RUNS } = require("./constants");

async function run() {
  for (let i = 0; i < RUNS; i++) {
    await fs.outputFile(FILE_PATH, TEST_STRING);
  }
}

run();
```

constants.js
```JS
const path = require("path");

const TARGET_DIR = path.join(__dirname, "test");
const FILE_PATH = path.join(TARGET_DIR, "test.js");
const TEST_STRING = "test".repeat(100000);
const RUNS = 7500;

module.exports = { TARGET_DIR, FILE_PATH, TEST_STRING, RUNS };
```

![Screenshot 2019-05-31 at 20 23 17](https://user-images.githubusercontent.com/2175521/58726466-0fa79800-83e2-11e9-9bea-b1c2f6c83e94.png)

Reply:
Thanks so much for looking into this @DeMoorJasper. Just removed the `existsCache` 😄